texthero/preprocessing.py

```diff
@@ -457,13 +472,67 @@ def clean(s: pd.Series, pipeline=None) -> pd.Series:
     """
 
     if not pipeline:
-        pipeline = get_default_pipeline()
+        return _optimised_default_clean(s)
 
     for f in pipeline:
         s = s.pipe(f)
     return s
 
 
+def _optimised_default_clean(s: pd.Series) -> pd.Series:
+    """
+    Applies the default clean pipeline in an optimised way to a series,
+    that is about 30% faster.
+
+    Default pipeline:
+     1. :meth:`texthero.preprocessing.fillna`
+     2. :meth:`texthero.preprocessing.lowercase`
+     3. :meth:`texthero.preprocessing.remove_digits`
+     4. :meth:`texthero.preprocessing.remove_punctuation`
+     5. :meth:`texthero.preprocessing.remove_diacritics`
+     6. :meth:`texthero.preprocessing.remove_stopwords`
+     7. :meth:`texthero.preprocessing.remove_whitespace`
+    """
+    return s.apply(_optimised_default_clean_single_cell)
+
+
+def _optimised_default_clean_single_cell(text: str) -> str:
+    """
+    Applies the default clean pipeline to one cell.
+
+    Default pipeline:
+     1. :meth:`texthero.preprocessing.fillna`
+     2. :meth:`texthero.preprocessing.lowercase`
+     3. :meth:`texthero.preprocessing.remove_digits`
+     4. :meth:`texthero.preprocessing.remove_punctuation`
+     5. :meth:`texthero.preprocessing.remove_diacritics`
+     6. :meth:`texthero.preprocessing.remove_stopwords`
+     7. :meth:`texthero.preprocessing.remove_whitespace`
+    """
+
+    # fillna
+    if pd.isna(text):
+        return ""
+
+    # lowercase
+    text = text.lower()
+
+    # remove digits and punctuation
+    pattern_mixed_remove = DIGITS_BLOCK + "|" + PUNCTUATION
```

Comment:
Is there a more elegant way of chaining regex? What are the best practices there?

Reply:
There is the re library to do this kind of stuff a bit more elegant, but I read, that str.replace gives us a performance improvement. As in my opinion, it is also quite clear, what it does, I think, we can keep it that way